src/ring_theory/valuation/valuation_ring.lean

```diff
@@ -270,6 +278,135 @@ end
 
 section
 
+variables {R : Type*} [comm_ring R] [is_domain R] {K : Type*}
+variables [field K] [algebra R K] [is_fraction_ring R K]
+
+lemma iff_dvd_total :
```

Comment:
I think `R` should be an explicit variable for all these `iff_foo` lemmas.

Reply:
It was suggested to be implicit by @Vierkantor in a review above. I'm not sure which is better?